pilot/pkg/config/kube/gateway/backend_policies.go

```diff
@@ -0,0 +1,423 @@
+	allPolicies := krt.JoinCollection([]krt.Collection[BackendPolicy]{backendTrafficPolicies, backendTLSPolicies})
+	byTarget := krt.NewIndex(allPolicies, func(o BackendPolicy) []TypedNamedspacedName {
+		return []TypedNamedspacedName{o.Target}
+	})
+	indexOpts := append(opts.WithName("BackendPolicyByTarget"), krt.WithIndexCollectionFromString(func(s string) TypedNamedspacedName {
+		parts := strings.Split(s, "/")
+		if len(parts) != 3 {
+			panic("invalid TypedNamedspacedName: " + s)
+		}
+		return TypedNamedspacedName{
+			NamespacedName: types.NamespacedName{
+				Namespace: parts[1],
+				Name:      parts[2],
+			},
+			Kind: kind.FromString(parts[0]),
+		}
+	}))
+	merged := krt.NewCollection(
+		byTarget.AsCollection(indexOpts...),
+		func(ctx krt.HandlerContext, i krt.IndexObject[TypedNamedspacedName, BackendPolicy]) **config.Config {
+			svc := i.Key
+			// Sort so we can pick the oldest, which will win
+			pols := slices.SortFunc(i.Objects, func(a, b BackendPolicy) int {
+				if r := a.CreationTime.Compare(b.CreationTime); r != 0 {
+					return r
+				}
+				if r := cmp.Compare(a.Source.Namespace, b.Source.Namespace); r != 0 {
+					return r
+				}
+				return cmp.Compare(a.Source.Name, b.Source.Name)
+			})
+			_ = pols
+			tlsSet := false
+			lbSet := false
+			spec := &networking.DestinationRule{
+				Host:          fmt.Sprintf("%s.%s.svc.%v", svc.Name, svc.Namespace, domainSuffix),
+				TrafficPolicy: &networking.TrafficPolicy{},
+			}
+			parents := make([]string, 0, len(pols))
+			for _, pol := range pols {
+				parents = append(parents, fmt.Sprintf("%s/%s.%s", pol.Source.Kind, pol.Source.Namespace, pol.Source.Name))
+				if pol.TLS != nil {
+					if tlsSet {
+						// We only allow 1. TODO: report status if there are multiple
+						continue
+					}
+					tlsSet = true
+					spec.TrafficPolicy.Tls = pol.TLS
+				}
+				if pol.LoadBalancer != nil {
+					if lbSet {
+						// We only allow 1. TODO: report status if there are multiple
+						continue
+					}
+					lbSet = true
+					spec.TrafficPolicy.LoadBalancer = pol.LoadBalancer
+				}
+			}
+			cfg := &config.Config{
+				Meta: config.Meta{
+					GroupVersionKind: gvk.DestinationRule,
+					Name:             fmt.Sprintf("%s-%s", svc.Name, constants.KubernetesGatewayName),
+					Namespace:        svc.Namespace,
+					Annotations: map[string]string{
+						constants.InternalParentNames: strings.Join(parents, ","),
+					},
+				},
+				Spec: spec,
+			}
+			return &cfg
+		}, opts.WithName("BackendPolicyMerged")...)
+	return merged
+}
+
+func BackendTLSPolicyCollection(
+	tlsPolicies krt.Collection[*gatewayalpha3.BackendTLSPolicy],
+	references *ReferenceSet,
+	opts krt.OptionsBuilder,
+) (krt.StatusCollection[*gatewayalpha3.BackendTLSPolicy, gatewayalpha2.PolicyStatus], krt.Collection[BackendPolicy]) {
+	return krt.NewStatusManyCollection(tlsPolicies, func(ctx krt.HandlerContext, i *gatewayalpha3.BackendTLSPolicy) (
+		*gatewayalpha2.PolicyStatus,
```

Comment:
Does v1alpha3 not have the correct status types?

Reply:
it points to v1alpha2 for some reason. The package scheme in the experimental types seems to be `alpha$RANDOM`, they are all mixed arbitrarily (even across x and not)